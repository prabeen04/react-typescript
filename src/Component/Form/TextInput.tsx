import * as React from 'react'

export interface ITextInputProps {
    value: string;
    name: string;
    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
    defaultValue?: string;
    placeholder?: string;
    className?: string;
    style?: object
}
export default function TextInput(props: ITextInputProps) {
    const { name, value, defaultValue, placeholder, onChange, ...rest } = props;
    return (
        <>
            <input
                data-test='TextInput'
                name={name}
                value={value}
                defaultValue={defaultValue}
                placeholder={placeholder}
                onChange={onChange}
                {...rest}
            />
        </>
    )
}
