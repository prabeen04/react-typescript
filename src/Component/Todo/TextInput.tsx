import * as React from 'react'

interface ITextInputProps {
    value: string;
    name: string;
    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
    placeholder?: string;
    className?: string
    style?: object
}
export default function TextInput(props: ITextInputProps) {
    const { value, placeholder, onChange, ...rest } = props;
    return (
        <>
            <input
                name={name}
                value={value}
                placeholder={placeholder}
                onChange={onChange}
                {...rest}
            />
        </>
    )
}
