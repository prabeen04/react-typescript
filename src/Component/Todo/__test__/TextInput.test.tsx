import * as React from 'react';
import { shallow } from "enzyme";
import TextInput from '../TextInput';

describe('<TextInput />', () => {
    it('should match snapshot', () => {
        const props = {
            name: '',
            value: '',
            onChange: jest.fn()

        }
        const wrapper = shallow(<TextInput {...props} />)
        expect(wrapper).toMatchSnapshot()

    })
    it('should render counter', () => {
        const props = {
            name: '',
            value: '',
            onChange: jest.fn()

        }
        const wrapper = shallow(<TextInput {...props} />)
        expect(wrapper.find('[data-test="TextInput"]').length).toEqual(1)
    })
})