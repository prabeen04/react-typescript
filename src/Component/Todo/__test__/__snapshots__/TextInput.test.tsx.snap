// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    name=""
    onChange={[MockFunction]}
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <input
        name=""
        onChange={[MockFunction]}
        value=""
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "name": "",
        "onChange": [MockFunction],
        "placeholder": undefined,
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <input
          name=""
          onChange={[MockFunction]}
          value=""
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "",
          "onChange": [MockFunction],
          "placeholder": undefined,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
